docs/references/security/rust-canister-development-security-best-practices.md

```diff
@@ -183,43 +183,185 @@ A canister could be rendered unusable so it could never be upgraded again e.g. d
 
 ## Inter-Canister Calls and Rollbacks
 
-### Don’t panic after await and don’t lock shared resources across await boundaries
+### Message Execution Basics
+
+To understand the issues around async inter-canister calls, one needs to understand a few properties about message execution. This is also explained in the [community conversation on Security Best Practices](https://www.youtube.com/watch?v=PneRzDmf_Xw&list=PLuhDt1vhGcrez-f3I0_hvbwGZHZzkZ7Ng&index=2&t=4s). 
+
+We fist provide a few definitions. A _call_ is a canister's implementation of either an [update](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-call) or [query call](https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-query) that it exposes. For example, if the Rust CDK is used, these are usually annotated with `#[query]` or `#[update]`, respectively. A _message_ is a set of consecutive instructions that a subnet executes for a canister. We'll see in the following that a call can be split into several messages if inter-canister calls are made. The following properties are essential: 
```

Comment:
```suggestion
We fist provide a few definitions. A _call_ is a canister's implementation of either an [update](../ic-interface-spec.md#http-call) or [query call](../ic-interface-spec.md#http-query) that it exposes. For example, if the Rust CDK is used, these are usually annotated with `#[query]` or `#[update]`, respectively. A _message_ is a set of consecutive instructions that a subnet executes for a canister. We'll see in the following that a call can be split into several messages if inter-canister calls are made. The following properties are essential: 
```
relative links including file extension, please. Otherwise links don't work or get broken without detecting it.